src/scenic/simulators/carla/simulator.py

```diff
@@ -243,3 +397,120 @@ def destroy(self):
 
 		self.world.tick()
 		super().destroy()
+
+	def process_rgb_image(self, image, buffer):
+		image.convert(cc.Raw)
+		buffer.append(image)
+
+	def process_depth_image(self, image, buffer):
+		image.convert(cc.LogarithmicDepth)
+		buffer.append(image)
+
+	def process_semantic_image(self, image, buffer):
+		image.convert(cc.CityScapesPalette)
+		buffer.append(image)
+
+	def process_lidar_data(self, lidar_data, buffer):
+		buffer.append(lidar_data)
+
+	def process_radar_data(self, radar_data, buffer):
+		buffer.append(radar_data)
+
+	def save_recordings(self, save_dir):
+		if not self.record_sensors:
+			print('No recordings saved; turn on recordings for simulator to enable')
```

Comment:
If you want this to not be a fatal error, use `warnings.warn` instead; otherwise raise an exception.